frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetailsPage.tsx

```diff
@@ -38,6 +44,56 @@ export function PolicyTemplateDetailsPage() {
     .replace(':kind', kind)
     .replace(':templateName', templateName)
 
+  let templateClusterName = clusterName
+  let templateNamespace = clusterName
+
+  // Determine if the policy framework is deployed in hosted mode. If so, the policy template needs to be retrieved
+  // from the hosting cluster instead of the managed cluster.
+  for (const addon of managedClusterAddOns) {
+    if (addon.metadata.namespace !== clusterName) {
+      continue
+    }
+
+    if (addon.metadata.name !== 'governance-policy-framework') {
+      continue
+    }
+
+    if (addon.metadata.annotations?.['addon.open-cluster-management.io/hosting-cluster-name']) {
+      templateClusterName = addon.metadata.annotations['addon.open-cluster-management.io/hosting-cluster-name']
+      // open-cluster-management-agent-addon is the default namespace but it shouldn't be used for hosted mode.
+      templateNamespace = addon.spec.installNamespace || 'open-cluster-management-agent-addon'
```

Comment:
Per SonarCloud: https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=3542&id=open-cluster-management_console&open=AZACWMrNQMxzjpPQgioY

```suggestion
      templateNamespace = addon.spec.installNamespace ?? 'open-cluster-management-agent-addon'
```